### PR TITLE
25 improve typing

### LINKED ISSUE #25: Improve typing
We could improve the typing In order to avoid to have to export/import typings for every block.

- [x] In any *.d.ts file remove all `import` and `export` (if any import or export is in d.ts file, typescript can't auto load the types anymore) https://github.com/superhuit-agency/eca-vaud.ch/commit/f9130eddae2ada46f8059cb0f24696a6031145c5
- [x]  Improve the block typings: https://github.com/superhuit-agency/eca-vaud.ch/commit/d5bdbe7cd0555d24050ad73723fec72ef05149f6
  - create a colocated typings.d.ts file
  - split attributes/props as they can differt from each other
  
- [x] Fix `tsconfig` in wordpress folder to also auto load typings https://github.com/superhuit-agency/eca-vaud.ch/commit/6f41492ed5aafbb195e56376dcfee841765fa1d9
- [x] Find a good way for typings to be correctly imported from `wordpress` folder into `src` folder (because I had to do this https://github.com/superhuit-agency/eca-vaud.ch/commit/9f5184bda9e985f6e6a5d2d1ecbaf787d64aa75f)

- [x] Add specific typing for Data & GraphQL for the `data.ts` file (e.g. https://github.com/superhuit-agency/eca-vaud.ch/blob/upgrade-stack/src/components/organisms/sections/SectionNews/typings.d.ts)

### PR DESCRIPTION
**Pull request by kuuak**
Fixes #25

### CODE REVIEW

**Review by charl0tee — CHANGES_REQUESTED** [review]
@kuuak I think we should add some info about this in the documentation :
- about the `typings.d.ts` file when we list all the files inside a Block's folder
- about Typings in general among the project (not add `import`/`export` etc)

File: docs/create-blocks/create-blocks.md [hunk 1]
```diff
@@ -90,6 +92,51 @@ export const ButtonBlock: WpBlockType = {
...
+```
+
+### Best Practices
+1. Keep types as specific as possible to ensure type safety.
+2. Use union types for attributes that can have multiple types.
+3. Make use of optional properties (?) for non-required attributes or props.
+4. Consider using utility types like `Partial<T>` or `Pick<T, K>` when appropriate.
+5. By following these guidelines, you can ensure that your block's typing is accurate, comprehensive, and helpful for development.
+
+By following these guidelines, you can ensure that your block's typing is accurate, comprehensive, and helpful for development.
```
Comment (charl0tee) [inline docs/create-blocks/create-blocks.md:137]: @kuuak Petite répétition 🙈

File: docs/create-blocks/create-blocks.md [hunk 2]
```diff
@@ -90,6 +92,51 @@ export const ButtonBlock: WpBlockType = {
 
 > We've added a custom property `postTypes` for the blocks' settings to define on which WP Post Type it can be available.<br /> 📚 [Learn more on how we've handled blocks whitelisting in here.](../features/blocks-whitelisting.md)
 
+## 🏷️ Typing `typings.d.ts`
+
+When you create a new block, a TypeScript typing file  is automatically generated. This file provides a basic structure for your block's types, but it's important to note that you should modify and adapt these types according to the specific needs of your block.
+
+The generated typing file typically includes:
+- An Attributes interface defining the block's attributes.
+- A Props interface extending the Attributes interface and potentially including additional props.
```
Comment (charl0tee) [inline docs/create-blocks/create-blocks.md:101]: @kuuak Should we maybe specify a bit that "block's attributes" is for the WP block, and the props are for the frontend ?

**Review by charl0tee — COMMENTED** [review]
@kuuak Nice, well done ! 👏🏽  I let also our master of documentation (AKA @snugglejuice) check this ;)

File: docs/create-blocks/create-blocks.md (hunk 2 above)
Comment (kuuak) [reply docs/create-blocks/create-blocks.md]: Good point, I did specify a bit more

**Review by charl0tee — DISMISSED** [review]

**Pull request merged**